src/SpeedDial/SpeedDial.js

```diff
@@ -0,0 +1,221 @@
+    const currentFocus = document.activeElement;
+
+    // If not actions, FAB must be focused.
+    if (currentFocus.parentElement.parentElement !== actions) {
+      if ((key === 'up' || key === 'down') && this.props.open) {
+        event.preventDefault();
+        actions.firstChild.firstChild.focus();
+
+        // This determines which key focuses the next / previous action.
+        // For example, if a visually impaired user presses down to select the first action
+        // (i.e. following DOM ordering), down will select the next action, and up the previous.
+        this.setState({ nextKey: key });
+        this.setState({ prevKey: key === 'up' ? 'down' : 'up' });
+      }
+    } else if (key === this.state.prevKey) {
+      event.preventDefault();
+      if (currentFocus.parentElement.previousElementSibling) {
+        currentFocus.parentElement.previousElementSibling.firstChild.focus();
+      } else {
+        findDOMNode(this.fab).focus();
+      }
+    } else if (key === this.state.nextKey) {
+      event.preventDefault();
+      if (currentFocus.parentElement.nextElementSibling) {
+        currentFocus.parentElement.nextElementSibling.firstChild.focus();
+      }
+    } else if (key === 'esc') {
+      fab.focus();
+      this.props.onClose(event, key);
+    }
+    if (this.props.onKeyDown) {
+      this.props.onKeyDown(event, key);
+    }
+  };
+
+  render() {
+    const {
+      children: childrenProp,
+      classes,
+      className: classNameProp,
+      hidden,
+      icon: iconProp,
+      label,
+      onKeyDown,
+      onClose,
+      open,
+      ...other
+    } = this.props;
+
+    const rootClassName = classNames(classes.root, classNameProp);
+    const buttonClassName = classNames(classes.fab, { [classes.fabHidden]: hidden });
+    const fabIconClassName = classNames(classes.fabIcon, { [classes.fabIconOpen]: open });
+    const actionsClassName = classNames(classes.actions, { [classes.actionsClosed]: !open });
+
+    // Filter the label for valid id characters.
+    const id = label.replace(/^[^a-z]+|[^\w:.-]+/gi, '');
+
+    let totalValidChildren = 0;
+    React.Children.forEach(childrenProp, child => {
+      if (React.isValidElement(child)) totalValidChildren += 1;
+    });
+
+    let validChildCount = 0;
+    const children = React.Children.map(childrenProp, child => {
+      if (!React.isValidElement(child)) return null;
+      const delay = 30 * (open ? validChildCount : totalValidChildren - validChildCount);
+      validChildCount += 1;
+      return React.cloneElement(child, {
+        delay,
+        open,
+        onKeyDown: this.handleKeyDown,
+        id: `${id}-item-${validChildCount}`,
+      });
+    });
+
+    let icon = null;
+    if (iconProp) {
+      if (React.isValidElement(iconProp)) {
+        icon = iconProp;
+      } else if (typeof iconProp === 'string') {
+        icon = <Icon>{iconProp}</Icon>;
+      }
+    } else {
+      icon = <AddIcon className={fabIconClassName} />;
```

Comment:
It would be nice to be able to use a different icon when hovered (e.g. a plus when not hovered and a pen icon when hovered, like in Google Inbox). Maybe by adding an `activeIcon` prop? :thinking:

Reply:
Good idea.